src/GovUk.Education.ExploreEducationStatistics.Common/Services/Interfaces/IStorageQueueService.cs

```diff
@@ -1,11 +1,16 @@
-﻿using System.Threading.Tasks;
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces
 {
     public interface IStorageQueueService
     {
         void AddMessages(string queueName, params object[] values);
```

Comment:
I know it's convenient, but wouldn't it be better to just scrap the params overload entirely in favour of the `IEnumerable` variant?

Reply:
Yeah I was really unsure about how to do this.

All but one of the usages is only sending a single message, so we could get rid of AddMessage**s** in favour of AddMessage and in the case of the Publisher, call it multiple times within a `for` for each Release that's been published. No future convenience of sending multiple messages with a single call.

We could do a minor refactor `AddMessages(params object[] values)` to `AddMessage(object value)` and add `AddMessages(IEnumerable<object> value)`. I think this is what you are suggesting but without refactoring all the other usages to wrap a single message in a list. This retains the convenience of sending multiple messages with a single call by using a list.

Or we have the implementation on this PR which has convenience of calling `AddMessages(message)`, AddMessages(message1, message2, message3)`, `AddMessages(listOfMessages)`.

Reply:
We should also be able to use our new `ForEachAsync` so I'm just trying that.

Reply:
Hmm, yeh I would say having that distinction between `AddMessage(message)` vs `AddMessages(listOfMessages)` is better for signalling intent in this case. 

I think the main problem here is that we're using param spread with `object` types. It could still be possible for someone to do `AddMessage(new List())` (instead of `AddMessages`) and that would still be valid. I wonder if a better solution would be to use a generic parameter for the service i.e. `IStorageQueueService<T>` and methods like `AddMessage(T message)`.

We could then have specific implementations which define what the messages should be and avoid most of these problems?

Reply:
I made a change to create the distinction between `AddMessage(message)` and `AddMessages(listOfMessages)`

At the moment it's caller beware I suppose, because someone might want to `AddMessage(list)` where they want a single message containing an array body.

I agree if we introduce the type parameter the intention is a bit clearer, e.g.  `StorageQueueService<List<MyType>>.AddMessage(list)` vs `StorageQueueService<MyType>.AddMessages(list) and the storage service can have a queue name and cleinet and not need to check the queue exists with every call. I think this is a bigger bit of refacoring though so we could create this as a new issue?

Reply:
Cool yep sounds good to me! :+1:

Reply:
Created https://dfedigital.atlassian.net/browse/EES-1554